internal/provider/data_logging_endpoints.go

```diff
@@ -0,0 +1,204 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"terraform-provider-altitude/internal/provider/client"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &LoggingEndpointsDataSource{}
+	_ datasource.DataSourceWithConfigure = &LoggingEndpointsDataSource{}
+)
+
+// NewLoggingEndpointsDataSource is a helper function to simplify the provider implementation.
+func NewLoggingEndpointsDataSource() datasource.DataSource {
+	return &LoggingEndpointsDataSource{}
+}
+
+// loggingEndpointsDataSource is the data source implementation.
+type LoggingEndpointsDataSource struct {
+	client *client.Client
+}
+
+// loggingEndpointsDataSourceModel maps the data source schema data.
+
+type LoggingEndpointsDataSourceModel struct {
+	Endpoints []LoggingEndpointDataSourceModel `tfsdk:"endpoints"`
+}
+type LoggingEndpointDataSourceModel struct {
+	ID            types.String                        `tfsdk:"id"`
+	Type          types.String                        `tfsdk:"type"`
+	EnvironmentId types.String                        `tfsdk:"environmentid"`
+	Config        GetAbstractAccessLoggingConfigModel `tfsdk:"config"`
+}
+
+type GetAbstractAccessLoggingConfigModel struct {
```

Comment:
Can we call this BQAccessLogging, it's not abstract. We can abstract it when it's required